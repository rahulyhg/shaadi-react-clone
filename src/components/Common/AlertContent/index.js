import React from 'react';
import PropTypes from 'prop-types';
import s from './styles';

const AlertMultiLine = props => (
  <s.AlertMultiLine size={props.alert.body.length}>
    {props.alert.title}
    {props.alert.body.map(para => (
      <span key={para.key}>
        {(para.body || para.items).map(content => {
          if (content.type === 'link') {
            return (
              <s.TooltipSpanLink key={content.key} to={content.url} isExternal={props.source === 'unified'}>
                {content.text}&nbsp;
              </s.TooltipSpanLink>
            );
          }
          if (content.type === 'button') {
            return (
              <div key="alert-content-btn">
                <s.TooltipSpanBtn key={content.key}>
                  <s.TooltipSpanLink key={content.key} to={content.url}>
                    {content.text}&nbsp;
                  </s.TooltipSpanLink>
                </s.TooltipSpanBtn>
              </div>
            );
          }
          return <s.TooltipSpanText key={content.key}>{content.text}&nbsp;</s.TooltipSpanText>;
        })}
      </span>
    ))}
  </s.AlertMultiLine>
);

const AlertContent = props => {
  if (props.type === 'toast' || props.type === 'flash') {
    return <AlertMultiLine alert={props.alert} source={props.source} />;
  }
  return <AlertMultiLine alert={props.alert} source={props.source} />;
};

const alertPropTypes = {
  key: PropTypes.string.isRequired,
  title: PropTypes.string,
  body: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      items: PropTypes.arrayOf(
        PropTypes.shape({
          key: PropTypes.string.isRequired,
          type: PropTypes.string.isRequired,
          text: PropTypes.string,
          url: PropTypes.string,
        }),
      ).isRequired,
    }),
  ).isRequired,
};

AlertContent.defaultProps = {
  source: '',
};
AlertMultiLine.defaultProps = {
  source: '',
};
AlertContent.propTypes = {
  type: PropTypes.oneOf(['toast', 'flash']).isRequired,
  alert: PropTypes.shape(alertPropTypes).isRequired,
  source: PropTypes.string,
};

AlertMultiLine.propTypes = {
  alert: PropTypes.shape(alertPropTypes).isRequired,
};

export default AlertContent;
