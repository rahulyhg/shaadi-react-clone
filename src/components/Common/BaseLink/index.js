import PropTypes from 'prop-types';
import React from 'react';
import s from './styles';

const BaseLink = props => (props.to ? <s.HyperLink {...props} /> : <s.HyperAnchor {...props} />);

BaseLink.defaultProps = {
  styleMixin: [],
  to: '',
};

BaseLink.propTypes = {
  styleMixin: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  to: PropTypes.string,
};
export default BaseLink;
