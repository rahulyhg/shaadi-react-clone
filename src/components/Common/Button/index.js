import PropTypes from 'prop-types';
import React from 'react';

const Button = props => {
  const { className, onClick, children, isVisible } = props;
  if (isVisible === false) {
    return <span />;
  }
  return <input type="button" value={children} onClick={onClick} className={className} />;
};

Button.defaultProps = {
  isVisible: true,
  className: '',
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  isVisible: PropTypes.bool,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
