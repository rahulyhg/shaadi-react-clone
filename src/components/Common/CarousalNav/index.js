import React from 'react';
import PropTypes from 'prop-types';
import './theme.css';

const config = {
  imagePath: '/assets/',
  left: '<',
  right: '>',
};
const CarousalNav = props => (
  <button className={`${props.className} ${props.theme} arrow_${props.direction} `} style={{ ...props.style }} onClick={props.onClick}>
    {props.arrowType === 'image' ? (
      <img alt="nav" src={`${config.imagePath}${props.arrowName}-${props.direction}.png`} />
    ) : (
      config[props.direction]
    )}
  </button>
);

CarousalNav.defaultProps = {
  theme: 'default',
  direction: 'right',
  className: '',
  style: {},
  arrowType: 'default',
  arrowName: 'nav',
};
CarousalNav.propTypes = {
  theme: PropTypes.string,
  className: PropTypes.string,
  direction: PropTypes.oneOf(['left', 'right']),
  style: PropTypes.shape({}),
  onClick: PropTypes.func.isRequired,
  arrowType: PropTypes.oneOf(['image', 'default']),
  arrowName: PropTypes.string,
};
export default CarousalNav;
