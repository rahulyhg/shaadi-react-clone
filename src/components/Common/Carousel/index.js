import React from 'react';
import PropTypes from 'prop-types';

import s from './styles';
import Wrapper from './Wrapper';

class Carousel extends React.PureComponent {
  constructor(props) {
    super(props);
    this.maxComponentsInFrame = props.maxComponentsInFrame ? props.maxComponentsInFrame : 3;
    this.totalSlides = props.slidesCnt ? props.slidesCnt : props.children.length;
    this.totalFrames = this.calculateFrames(props);
    this.slideWidth = props.stepsSize || props.width / this.maxComponentsInFrame;
    this.steps = props.steps;
    this.state = {
      activeSlide: this.maxComponentsInFrame,
      activeFrame: 1,
      left: 0,
      shouldHidePrev: true,
      shouldHideNext: this.totalSlides <= this.maxComponentsInFrame,
    };
    this.maxLeftMargin = -(this.totalSlides - this.maxComponentsInFrame) * this.slideWidth;
    this.navSlide = this.navSlide.bind(this);
  }

  calculateFrames(props) {
    return this.totalSlides % this.maxComponentsInFrame === 0
      ? Math.floor(this.totalSlides / this.maxComponentsInFrame)
      : Math.ceil(this.totalSlides / this.maxComponentsInFrame);
  }

  navSlide(direction) {
    const updateState = {};
    const navWidth = this.slideWidth * this.steps;
    this.props.onAction(null, 'closeAllTooltips');
    switch (direction) {
      case 'next': {
        if (this.state.activeSlide < this.totalSlides) {
          const activeSlide = parseFloat((this.state.activeSlide + this.steps).toFixed(2));
          updateState.activeSlide = activeSlide;
          const left = this.state.left - navWidth;
          updateState.left = left < this.maxLeftMargin ? this.maxLeftMargin : left;
          updateState.shouldHideNext = activeSlide >= this.totalSlides;
          updateState.shouldHidePrev = activeSlide <= 0;
          if (!this.props.isSuccessStory) {
            window.scroll(window.scrollX + 1, window.scrollY - 1);
            window.scroll(window.scrollX - 1, window.scrollY + 1);
          }
        }
        break;
      }
      case 'prev': {
        if (this.state.activeSlide > this.maxComponentsInFrame) {
          const activeSlide = parseFloat((this.state.activeSlide - this.steps).toFixed(2));
          updateState.activeSlide = activeSlide;
          const left = this.state.left + navWidth;
          updateState.left = left > 0 ? 0 : left;
          updateState.shouldHideNext = activeSlide >= this.totalSlides;
          updateState.shouldHidePrev = activeSlide <= this.maxComponentsInFrame;
        }
        break;
      }
      // no default
    }
    if (Object.keys(updateState).length) {
      this.setState(updateState);
    }
  }
  renderCarousalBtn = (source, direction) => {
    const shouldHide = direction === 'prev' ? this.state.shouldHidePrev : this.state.shouldHideNext;
    const props = {
      hide: shouldHide,
      direction,
      isSuccessStory: this.props.isSuccessStory,
      isSimilar: source === 'similarProfile',
      isHovered: this.props.isHovered,
    };
    props[`${direction === 'prev' ? 'data-prevdisplay' : 'data-nextdisplay'}`] = !shouldHide;
    switch (source) {
      case 'similarProfile': {
        return (
          <s.RoundNavInnerWrap {...props}>
            <s.RoundNavInner direction={direction} onClick={() => this.navSlide(direction)} />
          </s.RoundNavInnerWrap>
        );
      }
      case 'premiumCarousel': {
        return (
          <s.RoundNavInnerWrap {...props}>
            <s.RoundNavInner direction={direction} hide={shouldHide} onClick={() => this.navSlide(direction)} />
          </s.RoundNavInnerWrap>
        );
      }
      case 'inbox': {
        return <s.RoundNav {...props} onClick={() => this.navSlide(direction)} />;
      }
      default: {
        return <s.ButtonText {...props} onClick={() => this.navSlide(direction)} />;
      }
    }
  };

  render() {
    return (
      <s.Carousel>
        <s.CarouseButton height={`${this.props.height}px`} source={this.props.source} hide={this.state.shouldHidePrev}>
          {this.renderCarousalBtn(this.props.source, 'prev')}
        </s.CarouseButton>
        {<s.shadow navDirection="prev" isVisible={this.props.source === 'inbox'} />}
        <s.OuterContainer
          listBucket={this.props.listBucket}
          premiumCarousel={this.props.premiumCarousel}
          height={`${this.props.height}px`}
          width={`${this.props.width}px`}
          isSuccessStory={this.props.isSuccessStory}
          source={this.props.source}
        >
          <Wrapper source={this.props.source} left={`${this.state.left}px`}>
            {this.props.children}
          </Wrapper>
        </s.OuterContainer>
        {<s.shadow navDirection="next" isVisible={this.props.source === 'inbox'} />}
        <s.CarouseButton height={`${this.props.height}px`} source={this.props.source} hide={this.state.shouldHideNext}>
          {this.renderCarousalBtn(this.props.source, 'next')}
        </s.CarouseButton>
      </s.Carousel>
    );
  }
}
Carousel.defaultProps = {
  premiumCarousel: false,
  listBucket: 'A',
  isSuccessStory: false,
  source: 'default',
  stepsSize: 0,
  isHovered: false,
};
Carousel.propTypes = {
  children: PropTypes.node.isRequired,
  onAction: PropTypes.func.isRequired,
  maxComponentsInFrame: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  slidesCnt: PropTypes.number.isRequired,
  steps: PropTypes.number.isRequired,
  premiumCarousel: PropTypes.bool,
  listBucket: PropTypes.string,
  isSuccessStory: PropTypes.bool,
  source: PropTypes.string,
  stepsSize: PropTypes.number,
  isHovered: PropTypes.bool,
};
export default Carousel;
