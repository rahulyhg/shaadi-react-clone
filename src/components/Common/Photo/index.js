import ImportFromFbBtn from './ImportFromFbBtn';
import UploadFromComputerBtn from './UploadFromComputerBtn';

export { UploadFromComputerBtn, ImportFromFbBtn };
