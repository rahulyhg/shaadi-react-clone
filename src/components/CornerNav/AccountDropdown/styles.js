import styled from 'styled-components';
import Link from '../../Common/Link';

const styles = {};

styles.DropdownWrapper = styled.div`
  position: relative;
`;

styles.PhotoUploadBtnsSpace = styled.div``;

styles.AccountDropdown = styled.div`
  display: ${props => (props.isVisible ? 'flex' : 'none')};
  flex-direction: column;
  position: absolute;
  right: 0;
  top: 56px;
  z-index: 1000;
  padding: 0;
  width: 390px;
  font-size: 13px;
  background: #fff;
  color: #72727d;
  background-clip: padding-box;
  cursor: text;
  border-radius: 3px;
  box-shadow: 0 4px 12px rgba(43, 59, 93, 0.35);
`;

styles.Thumbnail = styled.img`
  display: inline-block;
  vertical-align: middle;
  width: 32px;
  height: 32px;
  border-radius: 50%;
`;

styles.DropdownLink = styled.div`
  position: relative;
  display: block;
  padding: 12px 10px;
  transition: color 0.2s ease;
  text-decoration: none;
  color: #72727d;
  cursor: pointer;
  line-height: ${props => (props.isProfileDropddown ? '12px' : '28px')};
  color: #fff;
  font-size: 16px;
  background-color: ${props => (props.isActive ? '#f14d53' : 'transparent')};

  &:hover {
    background-color: #f14d53;
    border-bottom-color: transparent;
  }
`;

styles.DropdownArrowIcon = styled.span`
  display: inline-block;
  width: 10px;
  height: 6px;
  margin-left: 7px;
  background-image: url(/assets/top-navabc-pro-help-icn.png);
`;

styles.PhotoUpload = styled.div`
  display: ${props => (props.isVisible ? 'block' : 'none')};
  background: #f4f4f7;
  padding: 16px 21px 13px;
  border-radius: 3px 3px 0 0;
`;

styles.PhotoUploadPrompt = styled.div`
  background: url(/assets/privacy-case-v3.png) no-repeat left -138px;
  font-size: 14px;
  padding: 0 0 2px 27px;
  line-height: 16px;
`;

styles.PhotoUploadBtns = styled.div`
  display: flex;
  padding: 8px 0 5px;
`;

styles.PhotoPrivateMsg = styled.div`
  font-size: 12px;
`;

styles.Links = styled.div`
  display: flex;
  padding: 9px 16px 8px;
  flex-wrap: wrap;
`;

styles.Link = styled(Link)`
  display: block;
  flex: 0 0 auto;
  width: 50%;
  color: #72727d;
  font-size: 12px;
  padding: 7px;
  box-sizing: border-box;
  text-decoration: none;

  &:hover {
    background: #f0f0f0;
    text-decoration: none;
    color: #72727d;
  }
`;

styles.Upgrade = styled.div`
  display: 'block';
  margin: 0 16px 0;
  border-top: 1px solid #dfe0e3;
  padding: 11px 0 0;
  text-align: center;
`;

styles.ExpiryDate = styled.span`
  display: ${props => (props.isVisible ? 'inline-block' : 'none')};
  margin-left: 10px;
  padding-left: 10px;

  &:before {
    background: url(https://img2.shaadi.com/assests/2016/images/small-badge-sprite-v3.png) no-repeat left -70px;
    width: 1px;
    height: 13px;
    display: inline-block;
    margin: 0 10px;
  }
`;

styles.Value = styled.span`
  font-weight: bold;
  color: #ff5a60;
`;

styles.CompareLink = styled(Link)`
  display: block;
  margin: 8px 0 14px;
  padding: 4px 10px;
  color: #00bcd5;
  text-decoration: none;
  outline: none;

  &:hover {
    text-decoration: underline;
  }
`;

styles.UpgradeBtn = styled(Link)`
  display: inline-block;
  font-size: 14px;
  padding: 4px 20px 5px;
  font-weight: bold;
  cursor: pointer;
  text-decoration: none;
  color: #fff;
  background: #00bcd5;
  border: 1px solid #00bcd5;
  border-radius: 3px;
  margin-top: 8px;
  &:hover {
    background: #0194a8;
    border: 1px solid #0194a8;
  }
`;

styles.LinkIcon = styled.div`
  display: inline-block;
  vertical-align: middle;
  width: 16px;
  height: 16px;
  padding: 0 0 0 12px;
  background-image: url(/assets/privacy-case-v3.png);
  background-repeat: no-repeat;
  background-position: left
    ${props =>
      ({
        my_profiles_male: '-206px',
        my_profiles_female: '-224px',
        account_settings: '-242px',
        contact_filters: '-258px',
        sms_alerts: '-272px',
        my_settings: '-289px',
        logout: '-307px',
      }[props.icon])};
`;

export default styles;
