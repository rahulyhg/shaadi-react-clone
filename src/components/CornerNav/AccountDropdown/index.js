import React from 'react';
import PropTypes from '../../../PropTypes';
import s from './styles';
import { UploadFromComputerBtn, ImportFromFbBtn } from '../../Common/Photo';

class AccountDropdown extends React.PureComponent {
  constructor(props) {
    super(props);
    this.onToggleDropdown = this.onToggleDropdown.bind(this);
    this.renderPhotoUploadSection = this.renderPhotoUploadSection.bind(this);
  }
  onToggleDropdown() {
    this.props.onVisibilityChange('account');
  }

  renderPhotoUploadSection() {
    return (
      <s.PhotoUpload isVisible={!this.props.settings.hasUploadedPhoto}>
        <s.PhotoUploadPrompt>Photo Upload Pending</s.PhotoUploadPrompt>
        <s.PhotoUploadBtns>
          <UploadFromComputerBtn padding="8px 24px 10px 10px" />
          <s.PhotoUploadBtnsSpace />
          <ImportFromFbBtn />
        </s.PhotoUploadBtns>
        <s.PhotoPrivateMsg>You can keep your Photos private.</s.PhotoPrivateMsg>
      </s.PhotoUpload>
    );
  }

  render() {
    const { props } = this;
    return (
      <s.DropdownWrapper>
        <s.DropdownLink id="profileDropDown" isProfileDropddown isActive={this.props.isOpen} onClick={this.onToggleDropdown}>
          <s.Thumbnail
            id="profileImgSmall"
            {...props.thumbnail.img}
            onContextMenu={() => false}
            onDragStart={() => false}
            onDragEnter={() => false}
            onDragOver={() => false}
            onDrop={() => false}
          />
          <s.DropdownArrowIcon />
        </s.DropdownLink>
        <s.AccountDropdown isVisible={this.props.isOpen}>
          {!this.props.settings.hasUploadedPhoto && this.renderPhotoUploadSection()}
          <s.Links>
            <s.Link to="/my-shaadi/profile" isExternal target="_self">
              <s.LinkIcon icon={props.settings.gender === 'Male' ? 'my_profiles_male' : 'my_profiles_female'} />
              My Profile
            </s.Link>
            <s.Link to="/my-shaadi/profile/alert" isExternal target="_self" rightWpar>
              <s.LinkIcon icon="sms_alerts" />
              Email / SMS Alerts
            </s.Link>
            <s.Link to="/my-shaadi/my-account" isExternal target="_self">
              <s.LinkIcon icon="account_settings" />
              Account Settings
            </s.Link>
            <s.Link to="/my-shaadi/my-account/privacy-settings?lnkref=TopSettingsLink" isExternal target="_self" rightWpar>
              <s.LinkIcon icon="my_settings" />
              Privacy Options
            </s.Link>
            <s.Link to="/my-shaadi/partner-profile/contact-filter" isExternal target="_self">
              <s.LinkIcon icon="contact_filters" />
              Contact Filters
            </s.Link>
            <s.Link to="/registration/user/logout" isExternal target="_self" rightWpar>
              <s.LinkIcon icon="logout" />
              Logout
            </s.Link>
          </s.Links>
          <s.Upgrade>
            <div>
              <span>
                Account Type:
                {props.accountType === 'FREE' ? <span>{' Free'}</span> : <s.Value> {props.plan}</s.Value>}
              </span>
              <s.ExpiryDate isVisible={props.accountType !== 'FREE'}>
                |&nbsp;&nbsp; Expiry:
                <s.Value> {props.expiryDate}</s.Value>
              </s.ExpiryDate>
            </div>

            <s.UpgradeBtn to="/payment?source=top_navbar_profiledropdown_upgrade" isExternal target="_blank" rel="noopener noreferrer">
              {props.upgradeType === 'extend' ? 'Extend Membership' : props.upgradeType === 'renew' ? 'Renew Membership' : 'Upgrade Now'}
            </s.UpgradeBtn>
            <s.CompareLink to="/my-shaadi/membership/compare-plans" target="_blank" rel="noopener noreferrer" isExternal>
              Compare memberships
            </s.CompareLink>
          </s.Upgrade>
        </s.AccountDropdown>
      </s.DropdownWrapper>
    );
  }
}

AccountDropdown.defaultProps = {
  accountType: 'FREE',
  plan: 'FREE',
  expiryDate: '',
};

const imgPropTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  title: PropTypes.string,
};

AccountDropdown.propTypes = {
  accountType: PropTypes.string,
  plan: PropTypes.string,
  expiryDate: PropTypes.string,
  isOpen: PropTypes.bool.isRequired,
  onVisibilityChange: PropTypes.func.isRequired,
  thumbnail: PropTypes.shape({ img: PropTypes.shape(imgPropTypes) }).isRequired,
  onAction: PropTypes.func.isRequired,
  settings: PropTypes.shape({
    hasUploadedPhoto: PropTypes.bool.isRequired,
  }).isRequired,
  upgradeType: PropTypes.string.isRequired,
};

export default AccountDropdown;
