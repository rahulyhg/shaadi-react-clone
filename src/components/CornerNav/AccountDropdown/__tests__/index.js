import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import AccountDropdown from '../index';
import initializeStore from '../../../../store';
import factory from './utils/factory';
import { UploadFromComputerBtn, ImportFromFbBtn } from '../../../Common/Photo';

jest.mock('../../../Common/Link');

const store = initializeStore();
const onAction = jest.fn();
const onDragStart = jest.fn();

describe('AccountDropdown Tests', () => {
  beforeEach(() => {
    onAction.mockClear();
  });

  it('should have upload photo buttons if profile doesnt have photo', () => {
    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...factory.props} />
        </Router>
      </Provider>,
    );
    expect(sampleWrapper.find('ImportFromFbBtn').length).toEqual(1);
    expect(sampleWrapper.find(UploadFromComputerBtn).length).toEqual(1);
  });

  it('should not have photo upload buttons if profile has photo', () => {
    const settings = { ...factory.props.settings, hasUploadedPhoto: true };
    const hasPhoto = { ...factory.props, settings };

    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...hasPhoto} />
        </Router>
      </Provider>,
    );
    expect(sampleWrapper.find(ImportFromFbBtn).length).toEqual(0);
    expect(sampleWrapper.find(UploadFromComputerBtn).length).toEqual(0);
  });

  it('should have male profile icon against My profile link', () => {
    const settings = { ...factory.props.settings, gender: 'Male' };
    const profileGender = { ...factory.props, settings };

    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...profileGender} />
        </Router>
      </Provider>,
    );
    expect(sampleWrapper.find('div[icon="my_profiles_male"]').exists()).toBe(true);
  });

  it('should have female profile icon against My profile link', () => {
    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...factory.props} />
        </Router>
      </Provider>,
    );

    expect(sampleWrapper.find('div[icon="my_profiles_female"]').exists()).toBe(true);
  });

  it('should show account dropdown when dropdown is clicked', () => {
    const onVisibilityChange = jest.fn();
    const settings = { ...factory.props, onVisibilityChange };

    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...settings} />
        </Router>
      </Provider>,
    );

    const button = sampleWrapper.find('#profileDropDown').at(0);
    expect(onAction).not.toHaveBeenCalled();

    button.simulate('click');

    expect(onVisibilityChange).toHaveBeenCalled();
  });

  it('should have onContextMenu  and drag events attached to image', () => {
    const settings = { ...factory.props, onDragStart };

    const sampleWrapper = shallow(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...settings} />
        </Router>
      </Provider>,
    );

    // const button = sampleWrapper.find('#profileImgSmall').at(0);

    // button.simulate('click');
    expect(sampleWrapper).toMatchSnapshot();
  });

  it('should show Extend Membership  if upgradeType is extend', () => {
    const currentMembership = { ...factory.props, upgradeType: 'extend' };

    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...currentMembership} />
        </Router>
      </Provider>,
    );
    expect(sampleWrapper.text()).toContain('Extend Membership');
  });

  it('should show Renew Membership  if upgradeType is renew', () => {
    const currentMembership = { ...factory.props, upgradeType: 'renew' };

    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...currentMembership} />
        </Router>
      </Provider>,
    );
    expect(sampleWrapper.text()).toContain('Renew Membership');
  });

  it('should show Upgrade Now  if upgradeType is upgrade', () => {
    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...factory.props} />
        </Router>
      </Provider>,
    );
    expect(sampleWrapper.text()).toContain('Upgrade Now');
  });

  it('should show premium plan taken by user if user has premium membership', () => {
    const membershipType = { ...factory.props, accountType: 'PAID', plan: 'Gold' };

    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...membershipType} />
        </Router>
      </Provider>,
    );

    expect(sampleWrapper.text()).toContain(`Account Type: ${membershipType.plan}`);
  });

  it('should show Free plan if user has no premium membership', () => {
    const sampleWrapper = mount(
      <Provider store={store}>
        <Router>
          <AccountDropdown {...factory.props} />
        </Router>
      </Provider>,
    );

    expect(sampleWrapper.text()).toContain(`Account Type: ${factory.props.plan}`);
  });
});
