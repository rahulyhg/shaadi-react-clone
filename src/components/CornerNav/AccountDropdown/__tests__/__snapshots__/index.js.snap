// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDropdown Tests should have onContextMenu  and drag events attached to image 1`] = `
<BrowserRouter>
  <AccountDropdown
    accountType="FREE"
    daysToExpiry={0}
    expiryDate=""
    isOpen={false}
    onAction={[Function]}
    onContextMenu={[Function]}
    onDragEnter={[Function]}
    onDragOver={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onVisibilityChange={[Function]}
    plan="Free"
    settings={
      Object {
        "canAccessChat": true,
        "canConnectWithMessage": false,
        "canInitiateChat": false,
        "canSendPasswordOnConnect": false,
        "canViewCollegeAndEmployer": false,
        "canViewHoroscope": true,
        "contactsRemaining": null,
        "contactsTotal": null,
        "defaultSearchFormat": "list",
        "experiments": Object {
          "near_me": Object {
            "bucket": "B",
          },
          "premium_carousel": Object {
            "bucket": "A",
          },
          "premium_tag": Object {
            "bucket": "A",
          },
          "webCTA_Dec2017": Object {
            "bucket": "B",
          },
        },
        "expiryDate": "",
        "gender": "Female",
        "hasUploadedPhoto": false,
        "horoscopeStyle": "l/ENG/hs/1",
        "isAstroGamified": true,
        "isBothPartyPayUser": false,
        "isFamilyGamified": true,
        "isHidden": false,
        "isIndianDiaspora": true,
        "isMobileVerified": false,
        "isNri": false,
        "isPaidUser": false,
        "isUnderScreening": false,
        "membershipTags": "free",
        "mobileNumber": "9379732670",
        "offerCode": "octbd60b",
        "offer_details": Array [
          Object {
            "type": "perc",
            "value": 60,
          },
        ],
        "showUpgradeBanner": true,
        "showUpgradeLinks": true,
        "wasPaidUser": false,
      }
    }
    thumbnail={
      Object {
        "img": Object {
          "alt": "User menu",
          "src": "https://img2.shaadi.com//imgs/my-shaadi/ver2/60-add-ph-female-v2.gif",
        },
      }
    }
    upgradeType="upgrade"
  />
</BrowserRouter>
`;
